### PR TITLE
feat: moonbeam

### PR DESCRIPTION
**Pull request by leoslr**
## Description of the changes

Deployment on moonbeam.

Unrelated: Deployment of the batch contract on optimism - was missing.

### CODE REVIEW

File: packages/smart-contracts/scripts-create2/utils.ts [hunk 1]
```diff
@@ -10,7 +10,6 @@ export const create2ContractDeploymentList = [
   'EthereumProxy',
   'EthereumFeeProxy',
   'ERC20FeeProxy',
-  'Erc20ConversionProxy',
```
Comment (leoslr) [inline packages/smart-contracts/scripts-create2/utils.ts:13]: This contract is deployed through the old script.

**Review by KolevDarko — APPROVED** [review]

File: packages/currency/src/conversion-aggregators.ts [hunk 2]
```diff
@@ -49,6 +49,7 @@ const noConversionNetworks: AggregatorsMap = {
   avalanche: {},
   bsc: {},
   optimism: {},
+  moonbeam: {},
```
Comment (leoslr) [inline packages/currency/src/conversion-aggregators.ts:52]: I made a proposal in https://github.com/RequestNetwork/requestNetwork/pull/1003 to remove this

File: packages/currency/src/native.ts [hunk 3]
```diff
@@ -105,6 +105,12 @@ export const nativeCurrencies: Record<NativeCurrencyType, (NativeCurrency & { na
       name: 'Optimism Ether',
       network: 'optimism',
     },
+    {
```
Comment (benjlevesque) [inline packages/currency/src/native.ts:108]: this makes me wonder why we have native tokens in currency-api...

**Review by benjlevesque — APPROVED** [review]

File: packages/currency/src/native.ts (hunk 3 above)
Comment (benjlevesque) [reply packages/currency/src/native.ts:108]: oh you're right! no declarative don't make sense in here. But we should remove all detected then, it's duplicated

Comment (leoslr) [reply packages/currency/src/native.ts:108]: The detected aren't registered in currency-api

Comment (benjlevesque) [reply packages/currency/src/native.ts:108]: OK thanks

**Pull request merged**